### PR TITLE
doc(encrypt): Allow customization of the cipher

### PR DESCRIPTION
**Pull request by mantissahz**
Extend the volume-encryption.md to explain customization of volume encryption.

longhorn/longhorn#3353
https://github.com/longhorn/longhorn/issues/3285

Signed-off-by: James Lu <james.lu@suse.com>

### CODE REVIEW

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md [hunk 1]
```diff
@@ -22,6 +22,14 @@ More information about the available template parameters can be found in the [Ku
 
 Example secret your encryption keys are specified as part of the `CRYPTO_KEY_VALUE` parameter.
 We use `stringData` as type here so we don't have to base64 encoded before submitting the secret via `kubectl create`.
+
+Parameters `CRYPTO_KEY_CIPHER`, `CRYPTO_KEY_HASH`, `CRYPTO_KEY_SIZE` and `CRYPTO_PBKDF` provide the customization for volume encryption.
+- `CRYPTO_KEY_CIPHER`: Set the cipher specification algorithm string. Default value is `aes-xts-plain64` for LUKS.
```
Comment (innobead) [inline content/docs/1.4.0/advanced-resources/security/volume-encryption.md:27]:
```suggestion
- `CRYPTO_KEY_CIPHER`: Sets the cipher specification algorithm string. The default value is `aes-xts-plain64` for LUKS.
```

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md [hunk 2]
```diff
@@ -22,6 +22,14 @@ More information about the available template parameters can be found in the [Ku
...
+- `CRYPTO_KEY_HASH`: Specifies the passphrase hash for `open`. Default value is `sha256`.
```
Comment (innobead) [inline content/docs/1.4.0/advanced-resources/security/volume-encryption.md:28]:
```suggestion
- `CRYPTO_KEY_HASH`: Specifies the passphrase hash for `open`. The default value is `sha256`.
```

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md [hunk 3]
```diff
@@ -22,6 +22,14 @@ More information about the available template parameters can be found in the [Ku
...
+- `CRYPTO_KEY_SIZE`: Sets key size in bits and it must be a multiple of 8. Default value is `256`.
```
Comment (innobead) [inline content/docs/1.4.0/advanced-resources/security/volume-encryption.md:29]:
```suggestion
- `CRYPTO_KEY_SIZE`: Sets the key size in bits and it must be a multiple of 8. The default value is `256`.
```

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md [hunk 4]
```diff
@@ -22,6 +22,14 @@ More information about the available template parameters can be found in the [Ku
...
+- `CRYPTO_PBKDF`: Set Password-Based Key Derivation Function (PBKDF) algorithm for LUKS keyslot. Default value is `argon2i`.
```
Comment (innobead) [inline content/docs/1.4.0/advanced-resources/security/volume-encryption.md:30]:
```suggestion
- `CRYPTO_PBKDF`: Sets Password-Based Key Derivation Function (PBKDF) algorithm for LUKS keyslot. The default value is `argon2i`.
```

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md [hunk 5]
```diff
@@ -22,6 +22,14 @@ More information about the available template parameters can be found in the [Ku
 
 Example secret your encryption keys are specified as part of the `CRYPTO_KEY_VALUE` parameter.
 We use `stringData` as type here so we don't have to base64 encoded before submitting the secret via `kubectl create`.
+
+Parameters `CRYPTO_KEY_CIPHER`, `CRYPTO_KEY_HASH`, `CRYPTO_KEY_SIZE` and `CRYPTO_PBKDF` provide the customization for volume encryption.
```
Comment (innobead) [inline content/docs/1.4.0/advanced-resources/security/volume-encryption.md:26]:
```suggestion
Besides `CRYPTO_KEY_VALUE`, parameters `CRYPTO_KEY_CIPHER`, `CRYPTO_KEY_HASH`, `CRYPTO_KEY_SIZE`, and `CRYPTO_PBKDF` provide the customization for volume encryption.
```

**Review by innobead — CHANGES_REQUESTED** [review]

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md (hunk 2 above)
Comment (mantissahz) [reply content/docs/1.4.0/advanced-resources/security/volume-encryption.md]: Modified. Thanks.

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md (hunk 3 above)
Comment (mantissahz) [reply content/docs/1.4.0/advanced-resources/security/volume-encryption.md]: Modified. Thanks.

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md (hunk 4 above)
Comment (mantissahz) [reply content/docs/1.4.0/advanced-resources/security/volume-encryption.md]: Modified. Thanks.

File: content/docs/1.4.0/advanced-resources/security/volume-encryption.md (hunk 5 above)
Comment (mantissahz) [reply content/docs/1.4.0/advanced-resources/security/volume-encryption.md]: Modified. Thanks.

**Review by innobead — APPROVED** [review]

**Pull request merged**